### ISSUE TITLE
VST3 doesn't respond to MIDI controllers

### ISSUE THREAD

**Issue by mkruselj** [asb2m10/dexed#266]
Looks like current VST3 build of Dexed doesn't at all respond to modwheel, channel aftertouch, breath controller or foot controller, so all the settings from the PARM page simply don't work.

I bet this is because Pascal forgot to introduce "fake parameters" to receive these CCs inside VST3 (since VST3 works fundamentally differently from VST2 in this regard).

I bet this is also the reason behind #264 too.

**Comment by asb2m10** [comment]
You mean this: https://forums.steinberg.net/t/vst3-and-midi-cc-pitfall/201879 :(

I confirm, midi cc messages are simply ignored. Do you have an example of a opensource VST3 that uses JUCE that bypass this limitation ? The only workaround I see is to fork JUCE...

**Comment by mkruselj** [comment]
You could take a look at OB-Xd. MIDI learn and sending modwheel works just fine over there.

**Comment by asb2m10** [comment]
Ok fix #244 did more problem than good. Bumping to JUCE 6.0.7 and removing the manual flag seems to have fixed everything. Automation works and midi CC works also. Please confirm with the lastest build.

**Comment by mkruselj** [comment]
Hmmm nope, CCs still not working over here somehow - in Reaper or Studio One, or Live 10...

**Comment by mkruselj** [comment]
Curious... I've just built it myself from source and it works! Unsure why it didn't work with the .vst3 from the nightly page...

**Comment by asb2m10** [comment]
Found the issue: the pipeline uses VS2019, you probably built it with VS2017. Fix it for both VS version. Now the nightly should be fine.

**Comment by mkruselj** [comment]
Actually I use VS2019 as well. It did ask me to convert some stuff when opening the solution, so I did.

Oh also, when I changed configuration to Release, it did not want to build because it was hardcoded to an older version of Windows 10 SDK. I had to manually change to just 10.0 (latest installed version) in each project's properties (so, three times, for shared code, standalone and VST3), in order to build a Release version. Just FYI :)
